types/host_requirements.go

```diff
@@ -0,0 +1,89 @@
+package types
+
+import (
+	"fmt"
+	"runtime"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/srl-labs/containerlab/virt"
+)
+
+type HostRequirements struct {
+	SSSE3                     bool          `json:"ssse3,omitempty"`         // ssse3 cpu instruction
+	VirtRequired              bool          `json:"virt-required,omitempty"` // indicates that KVM virtualization is required for this node to run
+	MinVCPU                   int           `json:"min-vcpu,omitempty"`      // the minimum amount of vcpus this node requires
+	MinVCPUFailAction         FailBehaviour `json:"min-vcpu-fail-action,omitempty"`
+	MinFreeMemoryGb           int           `json:"min-free-memory,omitempty"` // The minimum amount of memory this node requires
+	MinFreeMemoryGbFailAction FailBehaviour `json:"min-free-memory-fail-action,omitempty"`
+}
+
+type FailBehaviour int
+
+const (
+	FailBehaviourLog FailBehaviour = iota
+	FailBehaviourError
+)
+
+// NewHostRequirements is the constructor for new HostRequirements structs
+func NewHostRequirements() *HostRequirements {
+	return &HostRequirements{
+		MinVCPUFailAction:         FailBehaviourLog,
+		MinFreeMemoryGbFailAction: FailBehaviourLog,
+	}
+}
+
+func (h *HostRequirements) Verify() error {
+	// check virtualization Support
+	if h.VirtRequired && !virt.VerifyVirtSupport() {
+		return fmt.Errorf("the CPU virtualization support is required, but not available")
+	}
+	// check SSSE3 support
+	if h.SSSE3 && !virt.VerifySSSE3Support() {
+		return fmt.Errorf("the SSSE3 CPU feature is required, but not available")
+	}
+	// check minimum vCPUs
+	if valid, num := h.verifyMinVCpu(); !valid {
+		message := fmt.Sprintf("the defined minimum vCPU amount based on the nodes in your topology is %d whilst only %d vCPUs are available", h.MinVCPU, num)
```

Comment:
I had a thought before, that the verify function should have access to the Node struct, so that we can say which node/kind puts this requirement.
But so far we can't do that, when the method belongs to the HostRequirements

```suggestion
		message := fmt.Sprintf("the topology requires minimum %d vCPUs, but the host has %d vCPUs", h.MinVCPU, num)
```